File: server/models/gamemodes/deathmatch.py
from models.gamemode import GameMode


class DeathMatch(GameMode):
    """This model is a game mode."""

    def __init__(self, server):
        super().__init__(server)

        # Rules
        self.max_players = 10
        self.max_rounds = 1
        self.instant_respawn = True
        self.time_warmup = 20
        self.time_per_round = 300
        self.time_pre_round = 0

File: src/hypex-core/src/main/java/cloud/marchand/hypex/core/models/gamemode/DeathMatch.java
package cloud.marchand.hypex.core.models.gamemode;

import cloud.marchand.hypex.core.models.Game;

/**
 * DeathMatch game mode.
 * 
 * Auto-Respawn on
 * Friendly-fire on
 */
public class DeathMatch extends Game {

    @Override
    protected void initialize() {
        // Basic settings
        maxPlayers = 16;
        maxRounds = 1;
        autoRespawn = true;
        friendlyFire = true;

        // Durations
        gameStartingDuration = 20l;
        roundDuration = 120l;
        roundEndedDuration = 20l;
        gameEndedDuration = 60l;
    }

}
